### ISSUE TITLE
conda build is broken

### ISSUE THREAD

**Issue by joelvbernier** [HEXRD/hexrd#29]
@cjh1 looks like the conda build is broken?  just tried the latest with a tag 0.7.11  (and v0.7.11) and got

```
Traceback (most recent call last):
  File "/Users/bernier2/miniconda3/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 469, in main
    execute(sys.argv[1:])
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 460, in execute
    verify=args.verify, variants=args.variants)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 2344, in build_tree
    notest=notest,
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 1335, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/render.py", line 747, in expand_outputs
    expanded_outputs[m.dist()] = (output_dict, m)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/metadata.py", line 1317, in dist
    return '%s-%s-%s' % (self.name(), self.version(), self.build_id())
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/metadata.py", line 1148, in version
    raise ValueError("Fully-rendered version can't start with period -  got %s", res)
ValueError: ("Fully-rendered version can't start with period -  got %s", '.7.11')
```

**Comment by cjh1** [comment]
@joelvbernier I will take a look

**Comment by joelvbernier** [comment]
Update -- deleted the local tags (the last tag was v0.7.10, but there are commits since then) and now the build works and gets number `hexrd-0.7.10-py38h39e3cac_14.tar.bz2`.  Any idea why the new tag barfed?  Should I keep the 'v' prefix?

**Comment by joelvbernier** [comment]
command was `conda build conda.recipe/ --python=3.8`.  The last number signifies 14 commits since v0.7.10 tag?

**Comment by cjh1** [comment]
I will take a look at let you know ...

**Comment by cjh1** [comment]
This is fixed by #30, now that we are not using a prefix, don't remove it!

**Comment by joelvbernier** [comment]
wait -- to be clear -- my tag '0.7.11' did not work, but 'v0.7.10' did.  So by "don't remove it" you mean keep the prefix?

**Comment by cjh1** [comment]
I mean the code shouldn't try to remove it! Don't add it :-) As in your next tag should be 0.7.11

**Comment by joelvbernier** [comment]
That is the tag (0.7.11) that gave me the error in the first place though, and I am up to date.  See below:

```
(hexrd_0.7) calhoun:hexrd3 bernier2$ git tag 0.7.11
(hexrd_0.7) calhoun:hexrd3 bernier2$ conda build conda.recipe/ --python=3.8
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Copying /Users/bernier2/Documents/GitHub/hexrd3 to /Users/bernier2/miniconda3/conda-bld/hexrd_1592426149940/work/
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from config.variant
INFO:conda_build.variants:Adding in variants from config.variant
Attempting to finalize metadata for hexrd
INFO:conda_build.metadata:Attempting to finalize metadata for hexrd
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Traceback (most recent call last):
  File "/Users/bernier2/miniconda3/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 469, in main
    execute(sys.argv[1:])
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 460, in execute
    verify=args.verify, variants=args.variants)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 2344, in build_tree
    notest=notest,
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 1335, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/render.py", line 747, in expand_outputs
    expanded_outputs[m.dist()] = (output_dict, m)
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/metadata.py", line 1317, in dist
    return '%s-%s-%s' % (self.name(), self.version(), self.build_id())
  File "/Users/bernier2/miniconda3/lib/python3.7/site-packages/conda_build/metadata.py", line 1148, in version
    raise ValueError("Fully-rendered version can't start with period -  got %s", res)
ValueError: ("Fully-rendered version can't start with period -  got %s", '.7.11')
```

**Comment by cjh1** [comment]
Are you running with my PR #30?

**Comment by joelvbernier** [comment]
ah, no, only up to #22 -- but versioneer is gone and the build strings are sane with the existing tag prefix; so what does #30 change?  I am getting `hexrd-0.7.11-py38h39e3cac_0.tar.bz2` with tag `v0.7.11`, which is 'good' behavior.

**Comment by cjh1** [comment]
#30 stops us remove the first character of the version string. As you are now tagging correctly with ```0.7.11``` rather than ```v0.7.11```. Without #30 a tag of ```0.7.11``` would give a version of ```.7.11``` hence the error above.

**Comment by joelvbernier** [comment]
merged #30

**Comment by joelvbernier** [comment]
Works, sorry for the confusion.
